### PR TITLE
PHP 8: Code Review `StudyProgramme`

### PR DESCRIPTION
**Pull request by mjansenDatabay**
Hi @klees ,

I completed the review of the `StudyProgramme` component.

Results:
- [ ] The code style is `PSR-2 + X` compliant
- [x] No usage of `$_GET` / `$_POST` / `$_REQUEST` / `$_SESSION`:
- [x] The unit tests pass
- [x] External libraries are compatible with PHP 8 (if relevant)
- [ ] There are no serious `PhpStorm Code Inspection` issues left
- [ ] The types are fully documented (PhpDoc) or (even better and a MUST if possible) explict PHP types are used (type hints, return types, typed properties)

Actions:
- [ ] Please check my inline comments. Some of them might be just information, some require an action. You can use `TODO PHP8-REVIEW` when searching.

Kind regards,
@mjansenDatabay

### CODE REVIEW

File: Modules/StudyProgramme/classes/model/Types/class.ilStudyProgrammeTypeDBRepository.php [hunk 1]
```diff
@@ -320,17 +320,18 @@ public function deleteType(ilStudyProgrammeType $type) : void
 
     protected function getActivePlugins() : Iterator
     {
-        return $this->component_factory->getActivePluginsInSlot('prgtypehk');;
+        return $this->component_factory->getActivePluginsInSlot('prgtypehk');
+        ;
```
Comment (tfamula) [inline Modules/StudyProgramme/classes/model/Types/class.ilStudyProgrammeTypeDBRepository.php]: Seems like one semicolon too much here ;)

Comment (mjansenDatabay) [reply Modules/StudyProgramme/classes/model/Types/class.ilStudyProgrammeTypeDBRepository.php]: Good catch

Comment (mjansenDatabay) [inline Modules/StudyProgramme/classes/model/Types/class.ilStudyProgrammeTypeDBRepository.php]:
```suggestion
```

**Comment by klees** [comment]
Changes applied via: https://github.com/ILIAS-eLearning/ILIAS/commit/a0e04554ab263f3901217b27307aa0cedf96344d

**Pull request closed without merging**